



<!DOCTYPE html>
<html lang="en" class="">
  <head prefix="og: http://ogp.me/ns# fb: http://ogp.me/ns/fb# object: http://ogp.me/ns/object# article: http://ogp.me/ns/article# profile: http://ogp.me/ns/profile#">
    <meta charset='utf-8'>

    <link crossorigin="anonymous" href="https://assets-cdn.github.com/assets/frameworks-06e4d8f59f5b927f13919c423f40f0888f37f9af7f426e4020948b00b9371059.css" media="all" rel="stylesheet" />
    <link crossorigin="anonymous" href="https://assets-cdn.github.com/assets/github-c1b086d5f5cc4f877f4b6ac865224ec861d335d78ee3471d96aae5c454471d48.css" media="all" rel="stylesheet" />
    
    
    <link crossorigin="anonymous" href="https://assets-cdn.github.com/assets/site-429acc2fc05149afec2a05196b3100560f117f753931d0afca46da6e1eeafe39.css" media="all" rel="stylesheet" />
    
    

    <link as="script" href="https://assets-cdn.github.com/assets/frameworks-f9a40617e5ea71342dc57103554b07a3c1edc88712b04de1a040aed44891a11d.js" rel="preload" />
    
    <link as="script" href="https://assets-cdn.github.com/assets/github-f0c4e3f3e6abd6591d9680dafdae890452bbeda80e6d138134b5c59eaa7cbca3.js" rel="preload" />

    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta http-equiv="Content-Language" content="en">
    <meta name="viewport" content="width=device-width">
    
    
    <title>Things noticed while porting wear from NW (TODO list) · Issue #243 · StephenBlackWasAlreadyTaken/xDrip-Experimental · GitHub</title>
    <link rel="search" type="application/opensearchdescription+xml" href="/opensearch.xml" title="GitHub">
    <link rel="fluid-icon" href="https://github.com/fluidicon.png" title="GitHub">
    <link rel="apple-touch-icon" href="/apple-touch-icon.png">
    <link rel="apple-touch-icon" sizes="57x57" href="/apple-touch-icon-57x57.png">
    <link rel="apple-touch-icon" sizes="60x60" href="/apple-touch-icon-60x60.png">
    <link rel="apple-touch-icon" sizes="72x72" href="/apple-touch-icon-72x72.png">
    <link rel="apple-touch-icon" sizes="76x76" href="/apple-touch-icon-76x76.png">
    <link rel="apple-touch-icon" sizes="114x114" href="/apple-touch-icon-114x114.png">
    <link rel="apple-touch-icon" sizes="120x120" href="/apple-touch-icon-120x120.png">
    <link rel="apple-touch-icon" sizes="144x144" href="/apple-touch-icon-144x144.png">
    <link rel="apple-touch-icon" sizes="152x152" href="/apple-touch-icon-152x152.png">
    <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon-180x180.png">
    <meta property="fb:app_id" content="1401488693436528">

      <meta content="https://avatars2.githubusercontent.com/u/9692866?v=3&amp;s=400" name="twitter:image:src" /><meta content="@github" name="twitter:site" /><meta content="summary" name="twitter:card" /><meta content="Things noticed while porting wear from NW (TODO list) · Issue #243 · StephenBlackWasAlreadyTaken/xDrip-Experimental" name="twitter:title" /><meta content="BgGraphBuilder - make many methods static or refactor them to a separate class (no need to load all data for the graph to unitize one value e.g.)
Play-Service versions for wear integration are pret..." name="twitter:description" />
      <meta content="https://avatars2.githubusercontent.com/u/9692866?v=3&amp;s=400" property="og:image" /><meta content="GitHub" property="og:site_name" /><meta content="object" property="og:type" /><meta content="Things noticed while porting wear from NW (TODO list) · Issue #243 · StephenBlackWasAlreadyTaken/xDrip-Experimental" property="og:title" /><meta content="https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/243" property="og:url" /><meta content="BgGraphBuilder - make many methods static or refactor them to a separate class (no need to load all data for the graph to unitize one value e.g.)
Play-Service versions for wear integration are pret..." property="og:description" />
      <meta name="browser-stats-url" content="https://api.github.com/_private/browser/stats">
    <meta name="browser-errors-url" content="https://api.github.com/_private/browser/errors">
    <link rel="assets" href="https://assets-cdn.github.com/">
    
    <meta name="pjax-timeout" content="1000">
    

    <meta name="msapplication-TileImage" content="/windows-tile.png">
    <meta name="msapplication-TileColor" content="#ffffff">
    <meta name="selected-link" value="repo_issues" data-pjax-transient>

    <meta name="google-site-verification" content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU">
<meta name="google-site-verification" content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA">
    <meta name="google-analytics" content="UA-3769691-2">

<meta content="collector.githubapp.com" name="octolytics-host" /><meta content="github" name="octolytics-app-id" /><meta content="4A58D104:337B:8F1B8B0:576D7158" name="octolytics-dimension-request_id" />
<meta content="/&lt;user-name&gt;/&lt;repo-name&gt;/issues/show" data-pjax-transient="true" name="analytics-location" />



  <meta class="js-ga-set" name="dimension1" content="Logged Out">



        <meta name="hostname" content="github.com">
    <meta name="user-login" content="">

        <meta name="expected-hostname" content="github.com">
      <meta name="js-proxy-site-detection-payload" content="ZmRkMzBhYmYyN2FmN2M4NDAxNjA0NGM1M2I0OGVmZDQzMThjZWY2YmMwMWY1YTE0MzY0YmI1YzE3ODNmYjg0ZHx7InJlbW90ZV9hZGRyZXNzIjoiNzQuODguMjA5LjQiLCJyZXF1ZXN0X2lkIjoiNEE1OEQxMDQ6MzM3Qjo4RjFCOEIwOjU3NkQ3MTU4IiwidGltZXN0YW1wIjoxNDY2NzkwMjMyfQ==">


      <link rel="mask-icon" href="https://assets-cdn.github.com/pinned-octocat.svg" color="#4078c0">
      <link rel="icon" type="image/x-icon" href="https://assets-cdn.github.com/favicon.ico">

    <meta name="html-safe-nonce" content="38931228b1d06c86a4f711348017e2f038669775">
    <meta content="2f8075894882f0879ba994800d0f8288ea12e966" name="form-nonce" />

    <meta http-equiv="x-pjax-version" content="b26dd6aa2b8d6a62805b37391f6299c4">
    

      
  <meta name="description" content="xDrip-Experimental - Experimental Branches for Collaboration on DexDrip">
  <meta name="go-import" content="github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental git https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental.git">

  <meta content="649942" name="octolytics-dimension-user_id" /><meta content="StephenBlackWasAlreadyTaken" name="octolytics-dimension-user_login" /><meta content="30898542" name="octolytics-dimension-repository_id" /><meta content="StephenBlackWasAlreadyTaken/xDrip-Experimental" name="octolytics-dimension-repository_nwo" /><meta content="true" name="octolytics-dimension-repository_public" /><meta content="false" name="octolytics-dimension-repository_is_fork" /><meta content="30898542" name="octolytics-dimension-repository_network_root_id" /><meta content="StephenBlackWasAlreadyTaken/xDrip-Experimental" name="octolytics-dimension-repository_network_root_nwo" />
  <link href="https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/commits/master.atom" rel="alternate" title="Recent Commits to xDrip-Experimental:master" type="application/atom+xml">


  </head>


  <body class="logged-out   env-production  vis-public">
    <div id="js-pjax-loader-bar" class="pjax-loader-bar"></div>
    <a href="#start-of-content" tabindex="1" class="accessibility-aid js-skip-to-content">Skip to content</a>

    
    
    



          <header class="site-header js-details-container" role="banner">
  <div class="container-responsive">
    <a class="header-logo-invertocat" href="https://github.com/" aria-label="Homepage" data-ga-click="(Logged out) Header, go to homepage, icon:logo-wordmark">
      <svg aria-hidden="true" class="octicon octicon-mark-github" height="32" version="1.1" viewBox="0 0 16 16" width="32"><path d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z"></path></svg>
    </a>

    <button class="btn-link right site-header-toggle js-details-target" type="button" aria-label="Toggle navigation">
      <svg aria-hidden="true" class="octicon octicon-three-bars" height="24" version="1.1" viewBox="0 0 12 16" width="18"><path d="M11.41 9H.59C0 9 0 8.59 0 8c0-.59 0-1 .59-1H11.4c.59 0 .59.41.59 1 0 .59 0 1-.59 1h.01zm0-4H.59C0 5 0 4.59 0 4c0-.59 0-1 .59-1H11.4c.59 0 .59.41.59 1 0 .59 0 1-.59 1h.01zM.59 11H11.4c.59 0 .59.41.59 1 0 .59 0 1-.59 1H.59C0 13 0 12.59 0 12c0-.59 0-1 .59-1z"></path></svg>
    </button>

    <div class="site-header-menu">
      <nav class="site-header-nav site-header-nav-main">
        <a href="/personal" class="js-selected-navigation-item nav-item nav-item-personal" data-ga-click="Header, click, Nav menu - item:personal" data-selected-links="/personal /personal">
          Personal
</a>        <a href="/open-source" class="js-selected-navigation-item nav-item nav-item-opensource" data-ga-click="Header, click, Nav menu - item:opensource" data-selected-links="/open-source /open-source">
          Open source
</a>        <a href="/business" class="js-selected-navigation-item nav-item nav-item-business" data-ga-click="Header, click, Nav menu - item:business" data-selected-links="/business /business/features /business/customers /business">
          Business
</a>        <a href="/explore" class="js-selected-navigation-item nav-item nav-item-explore" data-ga-click="Header, click, Nav menu - item:explore" data-selected-links="/explore /trending /trending/developers /integrations /integrations/feature/code /integrations/feature/collaborate /integrations/feature/ship /explore">
          Explore
</a>      </nav>

      <div class="site-header-actions">
            <a class="btn btn-primary site-header-actions-btn" href="/join?source=header-repo" data-ga-click="(Logged out) Header, clicked Sign up, text:sign-up">Sign up</a>
          <a class="btn site-header-actions-btn mr-2" href="/login?return_to=%2FStephenBlackWasAlreadyTaken%2FxDrip-Experimental%2Fissues%2F243" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in">Sign in</a>
      </div>

        <nav class="site-header-nav site-header-nav-secondary">
          <a class="nav-item" href="/pricing">Pricing</a>
          <a class="nav-item" href="/blog">Blog</a>
          <a class="nav-item" href="https://help.github.com">Support</a>
          <a class="nav-item header-search-link" href="https://github.com/search">Search GitHub</a>
              <div class="header-search scoped-search site-scoped-search js-site-search" role="search">
  <!-- </textarea> --><!-- '"` --><form accept-charset="UTF-8" action="/StephenBlackWasAlreadyTaken/xDrip-Experimental/search" class="js-site-search-form" data-scoped-search-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/search" data-unscoped-search-url="/search" method="get"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="&#x2713;" /></div>
    <label class="form-control header-search-wrapper js-chromeless-input-container">
      <div class="header-search-scope">This repository</div>
      <input type="text"
        class="form-control header-search-input js-site-search-focus js-site-search-field is-clearable"
        data-hotkey="s"
        name="q"
        placeholder="Search"
        aria-label="Search this repository"
        data-unscoped-placeholder="Search GitHub"
        data-scoped-placeholder="Search"
        tabindex="1"
        autocapitalize="off">
    </label>
</form></div>

        </nav>
    </div>
  </div>
</header>



    <div id="start-of-content" class="accessibility-aid"></div>

      <div id="js-flash-container">
</div>


    <div role="main" class="main-content">
        <div itemscope itemtype="http://schema.org/SoftwareSourceCode">
    <div id="js-repo-pjax-container" data-pjax-container>
      
<div class="pagehead repohead instapaper_ignore readability-menu experiment-repo-nav">
  <div class="container repohead-details-container">

    

<ul class="pagehead-actions">

  <li>
      <a href="/login?return_to=%2FStephenBlackWasAlreadyTaken%2FxDrip-Experimental"
    class="btn btn-sm btn-with-count tooltipped tooltipped-n"
    aria-label="You must be signed in to watch a repository" rel="nofollow">
    <svg aria-hidden="true" class="octicon octicon-eye" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M8.06 2C3 2 0 8 0 8s3 6 8.06 6C13 14 16 8 16 8s-3-6-7.94-6zM8 12c-2.2 0-4-1.78-4-4 0-2.2 1.8-4 4-4 2.22 0 4 1.8 4 4 0 2.22-1.78 4-4 4zm2-4c0 1.11-.89 2-2 2-1.11 0-2-.89-2-2 0-1.11.89-2 2-2 1.11 0 2 .89 2 2z"></path></svg>
    Watch
  </a>
  <a class="social-count" href="/StephenBlackWasAlreadyTaken/xDrip-Experimental/watchers">
    29
  </a>

  </li>

  <li>
      <a href="/login?return_to=%2FStephenBlackWasAlreadyTaken%2FxDrip-Experimental"
    class="btn btn-sm btn-with-count tooltipped tooltipped-n"
    aria-label="You must be signed in to star a repository" rel="nofollow">
    <svg aria-hidden="true" class="octicon octicon-star" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74z"></path></svg>
    Star
  </a>

    <a class="social-count js-social-count" href="/StephenBlackWasAlreadyTaken/xDrip-Experimental/stargazers">
      23
    </a>

  </li>

  <li>
      <a href="/login?return_to=%2FStephenBlackWasAlreadyTaken%2FxDrip-Experimental"
        class="btn btn-sm btn-with-count tooltipped tooltipped-n"
        aria-label="You must be signed in to fork a repository" rel="nofollow">
        <svg aria-hidden="true" class="octicon octicon-repo-forked" height="16" version="1.1" viewBox="0 0 10 16" width="10"><path d="M8 1a1.993 1.993 0 0 0-1 3.72V6L5 8 3 6V4.72A1.993 1.993 0 0 0 2 1a1.993 1.993 0 0 0-1 3.72V6.5l3 3v1.78A1.993 1.993 0 0 0 5 15a1.993 1.993 0 0 0 1-3.72V9.5l3-3V4.72A1.993 1.993 0 0 0 8 1zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm3 10c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm3-10c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"></path></svg>
        Fork
      </a>

    <a href="/StephenBlackWasAlreadyTaken/xDrip-Experimental/network" class="social-count">
      63
    </a>
  </li>
</ul>

    <h1 class="public ">
  <svg aria-hidden="true" class="octicon octicon-repo" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path d="M4 9H3V8h1v1zm0-3H3v1h1V6zm0-2H3v1h1V4zm0-2H3v1h1V2zm8-1v12c0 .55-.45 1-1 1H6v2l-1.5-1.5L3 16v-2H1c-.55 0-1-.45-1-1V1c0-.55.45-1 1-1h10c.55 0 1 .45 1 1zm-1 10H1v2h2v-1h3v1h5v-2zm0-10H2v9h9V1z"></path></svg>
  <span class="author" itemprop="author"><a href="/StephenBlackWasAlreadyTaken" class="url fn" rel="author">StephenBlackWasAlreadyTaken</a></span><!--
--><span class="path-divider">/</span><!--
--><strong itemprop="name"><a href="/StephenBlackWasAlreadyTaken/xDrip-Experimental" data-pjax="#js-repo-pjax-container">xDrip-Experimental</a></strong>

</h1>

  </div>
  <div class="container">
    
<nav class="reponav js-repo-nav js-sidenav-container-pjax"
     itemscope
     itemtype="http://schema.org/BreadcrumbList"
     role="navigation"
     data-pjax="#js-repo-pjax-container">

  <span itemscope itemtype="http://schema.org/ListItem" itemprop="itemListElement">
    <a href="/StephenBlackWasAlreadyTaken/xDrip-Experimental" class="js-selected-navigation-item reponav-item" data-hotkey="g c" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches /StephenBlackWasAlreadyTaken/xDrip-Experimental" itemprop="url">
      <svg aria-hidden="true" class="octicon octicon-code" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path d="M9.5 3L8 4.5 11.5 8 8 11.5 9.5 13 14 8 9.5 3zm-5 0L0 8l4.5 5L6 11.5 2.5 8 6 4.5 4.5 3z"></path></svg>
      <span itemprop="name">Code</span>
      <meta itemprop="position" content="1">
</a>  </span>

    <span itemscope itemtype="http://schema.org/ListItem" itemprop="itemListElement">
      <a href="/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues" aria-selected="true" class="js-selected-navigation-item selected reponav-item" data-hotkey="g i" data-selected-links="repo_issues repo_labels repo_milestones /StephenBlackWasAlreadyTaken/xDrip-Experimental/issues" itemprop="url">
        <svg aria-hidden="true" class="octicon octicon-issue-opened" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
        <span itemprop="name">Issues</span>
        <span class="counter">35</span>
        <meta itemprop="position" content="2">
</a>    </span>

  <span itemscope itemtype="http://schema.org/ListItem" itemprop="itemListElement">
    <a href="/StephenBlackWasAlreadyTaken/xDrip-Experimental/pulls" class="js-selected-navigation-item reponav-item" data-hotkey="g p" data-selected-links="repo_pulls /StephenBlackWasAlreadyTaken/xDrip-Experimental/pulls" itemprop="url">
      <svg aria-hidden="true" class="octicon octicon-git-pull-request" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path d="M11 11.28V5c-.03-.78-.34-1.47-.94-2.06C9.46 2.35 8.78 2.03 8 2H7V0L4 3l3 3V4h1c.27.02.48.11.69.31.21.2.3.42.31.69v6.28A1.993 1.993 0 0 0 10 15a1.993 1.993 0 0 0 1-3.72zm-1 2.92c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zM4 3c0-1.11-.89-2-2-2a1.993 1.993 0 0 0-1 3.72v6.56A1.993 1.993 0 0 0 2 15a1.993 1.993 0 0 0 1-3.72V4.72c.59-.34 1-.98 1-1.72zm-.8 10c0 .66-.55 1.2-1.2 1.2-.65 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"></path></svg>
      <span itemprop="name">Pull requests</span>
      <span class="counter">13</span>
      <meta itemprop="position" content="3">
</a>  </span>



  <a href="/StephenBlackWasAlreadyTaken/xDrip-Experimental/pulse" class="js-selected-navigation-item reponav-item" data-selected-links="pulse /StephenBlackWasAlreadyTaken/xDrip-Experimental/pulse">
    <svg aria-hidden="true" class="octicon octicon-pulse" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path d="M11.5 8L8.8 5.4 6.6 8.5 5.5 1.6 2.38 8H0v2h3.6l.9-1.8.9 5.4L9 8.5l1.6 1.5H14V8z"></path></svg>
    Pulse
</a>
  <a href="/StephenBlackWasAlreadyTaken/xDrip-Experimental/graphs" class="js-selected-navigation-item reponav-item" data-selected-links="repo_graphs repo_contributors /StephenBlackWasAlreadyTaken/xDrip-Experimental/graphs">
    <svg aria-hidden="true" class="octicon octicon-graph" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M16 14v1H0V0h1v14h15zM5 13H3V8h2v5zm4 0H7V3h2v10zm4 0h-2V6h2v7z"></path></svg>
    Graphs
</a>

</nav>

  </div>
</div>

<div class="container new-discussion-timeline experiment-repo-nav">
  <div class="repository-content">

    
<style type="text/css" media="screen">
  span.labelstyle-fc2929, .linked-labelstyle-fc2929 {  background-color: #fc2929 !important;  color: #fff !important;}.labelstyle-fc2929.selected {  background-color: #fc2929 !important;  color: #fff !important;}.label-select-menu .labelstyle-fc2929.selected {  background:rgba(252, 41, 41, 0.12) !important;  color: #991818 !important;}

span.labelstyle-cccccc, .linked-labelstyle-cccccc {  background-color: #cccccc !important;  color: #333333 !important;}.labelstyle-cccccc.selected {  background-color: #cccccc !important;  color: #333333 !important;}.label-select-menu .labelstyle-cccccc.selected {  background:rgba(204, 204, 204, 0.12) !important;  color: #999999 !important;}

span.labelstyle-84b6eb, .linked-labelstyle-84b6eb {  background-color: #84b6eb !important;  color: #1c2733 !important;}.labelstyle-84b6eb.selected {  background-color: #84b6eb !important;  color: #1c2733 !important;}.label-select-menu .labelstyle-84b6eb.selected {  background:rgba(132, 182, 235, 0.12) !important;  color: #557699 !important;}

span.labelstyle-159818, .linked-labelstyle-159818 {  background-color: #159818 !important;  color: #fff !important;}.labelstyle-159818.selected {  background-color: #159818 !important;  color: #fff !important;}.label-select-menu .labelstyle-159818.selected {  background:rgba(21, 152, 24, 0.12) !important;  color: #159918 !important;}

span.labelstyle-e6e6e6, .linked-labelstyle-e6e6e6 {  background-color: #e6e6e6 !important;  color: #333333 !important;}.labelstyle-e6e6e6.selected {  background-color: #e6e6e6 !important;  color: #333333 !important;}.label-select-menu .labelstyle-e6e6e6.selected {  background:rgba(230, 230, 230, 0.12) !important;  color: #999999 !important;}

span.labelstyle-cc317c, .linked-labelstyle-cc317c {  background-color: #cc317c !important;  color: #fff !important;}.labelstyle-cc317c.selected {  background-color: #cc317c !important;  color: #fff !important;}.label-select-menu .labelstyle-cc317c.selected {  background:rgba(204, 49, 124, 0.12) !important;  color: #99245c !important;}

span.labelstyle-ffffff, .linked-labelstyle-ffffff {  background-color: #ffffff !important;  color: #333333 !important;}.labelstyle-ffffff.selected {  background-color: #ffffff !important;  color: #333333 !important;}.label-select-menu .labelstyle-ffffff.selected {  background:rgba(255, 255, 255, 0.12) !important;  color: #999999 !important;}
</style>

<div class="issues-listing" data-pjax>
    <div id="show_issue" class="js-issues-results">

    
  <div
    id="partial-discussion-header"
    class="gh-header js-details-container js-socket-channel js-updatable-content issue"
    data-channel="tenant:1:issue:125486830"
    data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/243/show_partial?partial=issues%2Ftitle">

  <div class="gh-header-show ">
      <div class="gh-header-actions">
          <a href="/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/new" class="btn btn-sm btn-primary right" data-hotkey="c">
            New issue
          </a>
      </div>

    <h1 class="gh-header-title">
      <span class="js-issue-title">Things noticed while porting wear from NW (TODO list)</span>
      <span class="gh-header-number">#243</span>
    </h1>
  </div>


  <div class="flex-table gh-header-meta">
    <div class="flex-table-item">
        <div class="state state-open">
          <svg aria-hidden="true" class="octicon octicon-issue-opened" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
          Open
        </div>
    </div>
    <div class="flex-table-item flex-table-item-primary">
        <a href="/AdrianLxM" class="author">AdrianLxM</a>  opened this <span class="noun">Issue</span> <relative-time datetime="2016-01-07T21:20:28Z">Jan 7, 2016</relative-time>
        &middot; 17 comments
    </div>
  </div>
</div>


    <div id="discussion_bucket" class="clearfix">
      <div class="discussion-sidebar js-sticky">
        <div id="partial-discussion-sidebar"
  class="js-socket-channel js-updatable-content"
  data-channel="tenant:1:issue:125486830"
  data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/243/show_partial?partial=issues%2Fsidebar">
  <div class="discussion-sidebar-item sidebar-labels js-discussion-sidebar-item">
  <!-- </textarea> --><!-- '"` --><form accept-charset="UTF-8" action="/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/243?partial=issues%2Fsidebar%2Fshow%2Flabels" class="js-issue-sidebar-form" data-form-nonce="2f8075894882f0879ba994800d0f8288ea12e966" data-remote="true" method="post"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="&#x2713;" /><input name="_method" type="hidden" value="put" /><input name="authenticity_token" type="hidden" value="xH21AlEUK12CZHCgFk+rQomJjjetb87bsXpmoNGUcavDaGgaba907hLKCpM83y1L71f79/vzFySLpBgCDwxNTQ==" /></div>
      
  <h3 class="discussion-sidebar-heading">
    Labels
  </h3>


    <div class="labels css-truncate">
    None yet
</div>

</form></div>

  <div class="discussion-sidebar-item sidebar-milestone js-discussion-sidebar-item">
  <!-- </textarea> --><!-- '"` --><form accept-charset="UTF-8" action="/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/243/set_milestone?partial=issues%2Fsidebar%2Fshow%2Fmilestone" class="js-issue-sidebar-form" data-form-nonce="2f8075894882f0879ba994800d0f8288ea12e966" data-remote="true" method="post"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="&#x2713;" /><input name="_method" type="hidden" value="put" /><input name="authenticity_token" type="hidden" value="bXR4nXjsOXNSoaPgSB6BbImZpdP5/74Nl+bOA+8m4eJInr//b+x8ji2wdOEkvuGZ5Yxi4W2521iXas3KEpwosQ==" /></div>
    
  <h3 class="discussion-sidebar-heading">
    Milestone
  </h3>


      No milestone

</form></div>


 <div class="discussion-sidebar-item sidebar-assignee js-discussion-sidebar-item">
  <!-- </textarea> --><!-- '"` --><form accept-charset="UTF-8" action="/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/243?partial=issues%2Fsidebar%2Fshow%2Fassignees" class="js-issue-sidebar-form" data-form-nonce="2f8075894882f0879ba994800d0f8288ea12e966" data-remote="true" method="post"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="&#x2713;" /><input name="_method" type="hidden" value="put" /><input name="authenticity_token" type="hidden" value="nLbWOj5ZRmWZG7hXDYydfzOnz31VvDGxpmDBEVuWYnDCgP72+hUSeyo2EWSJyp3yT5vJuxLPUpAFyFejhzahNQ==" /></div>
    
  <h3 class="discussion-sidebar-heading">
    Assignees
  </h3>


    <span class="css-truncate">
    No one assigned
</span>

</form></div>



  <div id="partial-users-participants" class="discussion-sidebar-item">
  <div class="participation">

    <h3 class="discussion-sidebar-heading">
      3 participants
    </h3>
    <div class="participation-avatars">
        <a class="participant-avatar tooltipped tooltipped-n" aria-label="AdrianLxM" href="/AdrianLxM"><img alt="@AdrianLxM" class="avatar" height="26" src="https://avatars3.githubusercontent.com/u/9692866?v=3&amp;s=52" width="26" /> </a>
        <a class="participant-avatar tooltipped tooltipped-n" aria-label="mgranberry" href="/mgranberry"><img alt="@mgranberry" class="avatar" height="26" src="https://avatars3.githubusercontent.com/u/152687?v=3&amp;s=52" width="26" /> </a>
        <a class="participant-avatar tooltipped tooltipped-n" aria-label="tzachi-dar" href="/tzachi-dar"><img alt="@tzachi-dar" class="avatar" height="26" src="https://avatars2.githubusercontent.com/u/7942372?v=3&amp;s=52" width="26" /> </a>
    </div>
  </div>
</div>


  
  
</div>

      </div>

      <div class="discussion-timeline js-quote-selection-container ">

        <div class="js-discussion js-socket-channel" data-channel="tenant:1:marked-as-read:issue:125486830">
          

  <div class="timeline-comment-wrapper js-comment-container">
    <a href="/AdrianLxM"><img alt="@AdrianLxM" class="timeline-comment-avatar" height="48" src="https://avatars2.githubusercontent.com/u/9692866?v=3&amp;s=96" width="48" /></a>
    <div id="issue-125486830"
     class="comment previewable-edit timeline-comment js-comment js-task-list-container
                    
                     
                    collaborator-comment  "
     data-body-version="712c0e587a0b635e4acc14d8dae95db3">

  <div class="timeline-comment-header ">

    <div class="timeline-comment-actions">
    </div>

    <span class="timeline-comment-label">Collaborator</span>


  <div class="timeline-comment-header-text">

    <strong>
      <a href="/AdrianLxM" class="author">AdrianLxM</a> 
    </strong>

    commented

    <a href="#issue-125486830" class="timestamp"><relative-time datetime="2016-01-07T21:20:28Z">Jan 7, 2016</relative-time></a>

  </div>
</div>


  <div class="comment-content">

    <div class="edit-comment-hide">
      <div class="comment-body markdown-body markdown-format js-comment-body">
          <ul>
<li><p>BgGraphBuilder - make many methods static or refactor them to a separate class (no need to load all data for the graph to unitize one value e.g.)</p></li>
<li><p>Play-Service versions for wear integration are pretty old and versions for "app" and "wear" do not match.</p></li>
<li>
<p>Integrate raw values - remove redundency to calculate "Nightscout-raw" and NS/DexCom calibraion format for:</p>

<ul>
<li>local broadcast</li>
<li>wear integration</li>
<li>upload to Share</li>
<li>upload to NightScout</li>
</ul>
</li>
<li><p>Integrate charging mode that turns the screen by 90° when watch is in the night-stand charger.</p></li>
<li><p>NightWatch has Activities of module "mobile" in the "wear" manifest. (delete them)</p></li>
<li><p>better wakelocks when sending data to the watch (not just a 15 seconds timeout wakelock)</p></li>
<li><p>Make timestamps of type <code>long</code> not <code>double</code> (especially in former NightWatch parts but xDrip still is not free from that)</p></li>
<li><p>Maybe: triggering a resend to wear from the gui could start a background thread "earlier" (not just for sending like now, but also for preparing the data).</p></li>
<li><p>Helper Methods fragmented over 'WatchUpdaterService.java' and 'BgGraphBuilder.java'.</p></li>
<li><p>Quite a few utility methods are of type: (already marked some)</p></li>
</ul>

<div class="highlight highlight-source-java"><pre><span class="pl-k">if</span>(booleanExpression){
    <span class="pl-k">return</span> <span class="pl-c1">true</span>;
<span class="pl-k">else</span> {
    <span class="pl-k">return</span> <span class="pl-c1">false</span>;
}</pre></div>

<ul>
<li><p>PebbleSync has no wakelocks</p></li>
<li><p>Pebble has a "resend to watch" function on NightWatch but not on xDrip.</p></li>
</ul>
      </div>
    </div>

  </div>

    
<div class="js-socket-channel js-updatable-content comment-reactions  js-reactions-container"
    data-channel="tenant:1:reaction:issue:125486830"
    data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/reactions/issue/125486830">
</div>

</div>

  </div>
  
      <div class="discussion-item discussion-item-ref">
    <div class="discussion-item-header" id="ref-pullrequest-125476069">
      <span class="discussion-item-icon">
        <svg aria-hidden="true" class="octicon octicon-bookmark" height="16" version="1.1" viewBox="0 0 10 16" width="10"><path d="M9 0H1C.27 0 0 .27 0 1v15l5-3.09L10 16V1c0-.73-.27-1-1-1zm-.78 4.25L6.36 5.61l.72 2.16c.06.22-.02.28-.2.17L5 6.6 3.12 7.94c-.19.11-.25.05-.2-.17l.72-2.16-1.86-1.36c-.17-.16-.14-.23.09-.23l2.3-.03.7-2.16h.25l.7 2.16 2.3.03c.23 0 .27.08.09.23h.01z"></path></svg>
      </span>

      <img alt="@AdrianLxM" class="avatar" height="16" src="https://avatars1.githubusercontent.com/u/9692866?v=3&amp;s=32" width="16" />
      <a href="/AdrianLxM" class="author" truncate="true">AdrianLxM</a>
      

      referenced
      this issue
      <a class="timestamp" href="#ref-pullrequest-125476069">
        <relative-time datetime="2016-01-07T21:42:14Z">Jan 7, 2016</relative-time>
      </a>
    </div>

      <span class="state state-merged right">
          <svg aria-hidden="true" class="octicon octicon-git-pull-request" height="14" version="1.1" viewBox="0 0 12 16" width="10"><path d="M11 11.28V5c-.03-.78-.34-1.47-.94-2.06C9.46 2.35 8.78 2.03 8 2H7V0L4 3l3 3V4h1c.27.02.48.11.69.31.21.2.3.42.31.69v6.28A1.993 1.993 0 0 0 10 15a1.993 1.993 0 0 0 1-3.72zm-1 2.92c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zM4 3c0-1.11-.89-2-2-2a1.993 1.993 0 0 0-1 3.72v6.56A1.993 1.993 0 0 0 2 15a1.993 1.993 0 0 0 1-3.72V4.72c.59-.34 1-.98 1-1.72zm-.8 10c0 .66-.55 1.2-1.2 1.2-.65 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"></path></svg>
        Merged
      </span>



    <h3 class="discussion-item-ref-title">
      <a href="/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/242" class="title-link">
        Wear integration
        <span class="issue-num">#242</span>
</a>    </h3>
    
  </div>


  <div class="timeline-comment-wrapper js-comment-container">
    <a href="/mgranberry"><img alt="@mgranberry" class="timeline-comment-avatar" height="48" src="https://avatars2.githubusercontent.com/u/152687?v=3&amp;s=96" width="48" /></a>
    <div id="issuecomment-169823982"
     class="comment previewable-edit timeline-comment js-comment js-task-list-container
                    
                     
                    collaborator-comment  "
     data-body-version="655e9cbcaf5907b951687795d722c1a3">

  <div class="timeline-comment-header ">

    <div class="timeline-comment-actions">
    </div>

    <span class="timeline-comment-label">Collaborator</span>


  <div class="timeline-comment-header-text">

    <strong>
      <a href="/mgranberry" class="author">mgranberry</a> 
    </strong>

    commented

    <a href="#issuecomment-169823982" class="timestamp"><relative-time datetime="2016-01-07T22:17:03Z">Jan 7, 2016</relative-time></a>

  </div>
</div>


  <div class="comment-content">

    <div class="edit-comment-hide">
      <div class="comment-body markdown-body markdown-format js-comment-body">
          <p>The IntelliJ 16 EAP has a pretty cool "find duplicated code" analyzer that gives a side-by-side difference between two bits of code.  It might be useful to play around with it while doing this.</p>
      </div>
    </div>

  </div>

    
<div class="js-socket-channel js-updatable-content comment-reactions  js-reactions-container"
    data-channel="tenant:1:reaction:issue-comment:169823982"
    data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/reactions/issue-comment/169823982">
</div>

</div>

  </div>
  <div class="timeline-comment-wrapper js-comment-container">
    <a href="/tzachi-dar"><img alt="@tzachi-dar" class="timeline-comment-avatar" height="48" src="https://avatars3.githubusercontent.com/u/7942372?v=3&amp;s=96" width="48" /></a>
    <div id="issuecomment-169840568"
     class="comment previewable-edit timeline-comment js-comment js-task-list-container
                    
                     
                    collaborator-comment  "
     data-body-version="0b3dc398b2a27ce9bcaf91dc829a9a8b">

  <div class="timeline-comment-header ">

    <div class="timeline-comment-actions">
    </div>

    <span class="timeline-comment-label">Collaborator</span>


  <div class="timeline-comment-header-text">

    <strong>
      <a href="/tzachi-dar" class="author">tzachi-dar</a> 
    </strong>

    commented

    <a href="#issuecomment-169840568" class="timestamp"><relative-time datetime="2016-01-07T23:33:51Z">Jan 7, 2016</relative-time></a>

  </div>
</div>


  <div class="comment-content">

    <div class="edit-comment-hide">
      <div class="comment-body markdown-body markdown-format js-comment-body">
          <p>What do you mean by "Integrate raw values"?</p>
      </div>
    </div>

  </div>

    
<div class="js-socket-channel js-updatable-content comment-reactions  js-reactions-container"
    data-channel="tenant:1:reaction:issue-comment:169840568"
    data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/reactions/issue-comment/169840568">
</div>

</div>

  </div>
  <div class="timeline-comment-wrapper js-comment-container">
    <a href="/AdrianLxM"><img alt="@AdrianLxM" class="timeline-comment-avatar" height="48" src="https://avatars2.githubusercontent.com/u/9692866?v=3&amp;s=96" width="48" /></a>
    <div id="issuecomment-169842866"
     class="comment previewable-edit timeline-comment js-comment js-task-list-container
                    
                     
                    collaborator-comment  "
     data-body-version="4996527e2d3fdabbb94538b77e30621d">

  <div class="timeline-comment-header ">

    <div class="timeline-comment-actions">
    </div>

    <span class="timeline-comment-label">Collaborator</span>


  <div class="timeline-comment-header-text">

    <strong>
      <a href="/AdrianLxM" class="author">AdrianLxM</a> 
    </strong>

    commented

    <a href="#issuecomment-169842866" class="timestamp"><relative-time datetime="2016-01-07T23:47:42Z">Jan 7, 2016</relative-time></a>

  </div>
</div>


  <div class="comment-content">

    <div class="edit-comment-hide">
      <div class="comment-body markdown-body markdown-format js-comment-body">
          <p><a href="https://github.com/tzachi-dar" class="user-mention">@tzachi-dar</a> I disabled that the watch faces show raw. I will enable it once we have a uniform calculation method.</p>

<p>There were just about 100 downloads of the apk that supports raw and I will reintegrate it before we have a release of xDrip-Viewer. So people using this feature will still have it.</p>
      </div>
    </div>

  </div>

    
<div class="js-socket-channel js-updatable-content comment-reactions  js-reactions-container"
    data-channel="tenant:1:reaction:issue-comment:169842866"
    data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/reactions/issue-comment/169842866">
</div>

</div>

  </div>
  <div class="timeline-comment-wrapper js-comment-container">
    <a href="/AdrianLxM"><img alt="@AdrianLxM" class="timeline-comment-avatar" height="48" src="https://avatars2.githubusercontent.com/u/9692866?v=3&amp;s=96" width="48" /></a>
    <div id="issuecomment-169843172"
     class="comment previewable-edit timeline-comment js-comment js-task-list-container
                    
                     
                    collaborator-comment  "
     data-body-version="11d465da755b8c7b390294fc306714c3">

  <div class="timeline-comment-header ">

    <div class="timeline-comment-actions">
    </div>

    <span class="timeline-comment-label">Collaborator</span>


  <div class="timeline-comment-header-text">

    <strong>
      <a href="/AdrianLxM" class="author">AdrianLxM</a> 
    </strong>

    commented

    <a href="#issuecomment-169843172" class="timestamp"><relative-time datetime="2016-01-07T23:49:30Z">Jan 7, 2016</relative-time></a>

  </div>
</div>


  <div class="comment-content">

    <div class="edit-comment-hide">
      <div class="comment-body markdown-body markdown-format js-comment-body">
          <p><a href="https://github.com/tzachi-dar" class="user-mention">@tzachi-dar</a> <a href="http://www.nightscout.info/wiki/labs/raw-data-with-nightwatch">http://www.nightscout.info/wiki/labs/raw-data-with-nightwatch</a></p>
      </div>
    </div>

  </div>

    
<div class="js-socket-channel js-updatable-content comment-reactions  js-reactions-container"
    data-channel="tenant:1:reaction:issue-comment:169843172"
    data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/reactions/issue-comment/169843172">
</div>

</div>

  </div>
  <div class="timeline-comment-wrapper js-comment-container">
    <a href="/tzachi-dar"><img alt="@tzachi-dar" class="timeline-comment-avatar" height="48" src="https://avatars3.githubusercontent.com/u/7942372?v=3&amp;s=96" width="48" /></a>
    <div id="issuecomment-169843638"
     class="comment previewable-edit timeline-comment js-comment js-task-list-container
                    
                     
                    collaborator-comment  "
     data-body-version="ad18f3c08bc01c0f1545c7daa391aab4">

  <div class="timeline-comment-header ">

    <div class="timeline-comment-actions">
    </div>

    <span class="timeline-comment-label">Collaborator</span>


  <div class="timeline-comment-header-text">
      <a href="#issuecomment-169843638" title="This comment left via email reply." class="comment-type-icon">
        <svg aria-hidden="true" class="octicon octicon-mail-read" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path d="M6 5H4V4h2v1zm3 1H4v1h5V6zm5-.48V14c0 .55-.45 1-1 1H1c-.55 0-1-.45-1-1V5.52c0-.33.16-.63.42-.81L2 3.58V3c0-.55.45-1 1-1h1.2L7 0l2.8 2H11c.55 0 1 .45 1 1v.58l1.58 1.13c.27.19.42.48.42.81zM3 7.5L7 10l4-2.5V3H3v4.5zm-2 6l4.5-3-4.5-3v6zm11 .5l-5-3-5 3h10zm1-6.5l-4.5 3 4.5 3v-6z"></path></svg>
      </a>

    <strong>
      <a href="/tzachi-dar" class="author">tzachi-dar</a> 
    </strong>

    commented

    <a href="#issuecomment-169843638" class="timestamp"><relative-time datetime="2016-01-07T23:52:23Z">Jan 7, 2016</relative-time></a>

  </div>
</div>


  <div class="comment-content">

    <div class="edit-comment-hide">
      <div class="comment-body markdown-body email-format js-comment-body">
          <div class="email-fragment">I was more interested in the way that you want to handle raw...
what version do you propose?</div>
<span class="email-hidden-toggle"><a href="#">…</a></span><div class="email-hidden-reply" style="display:none">
<div class="email-quoted-reply">On Fri, Jan 8, 2016 at 1:49 AM, AdrianLxM ***@***.***&gt; wrote:
 <a href="https://github.com/tzachi-dar" class="user-mention">@tzachi-dar</a> &lt;<a href="https://github.com/tzachi-dar">https://github.com/tzachi-dar</a>&gt;
 <a href="http://www.nightscout.info/wiki/labs/raw-data-with-nightwatch">http://www.nightscout.info/wiki/labs/raw-data-with-nightwatch</a>

 —
 Reply to this email directly or view it on GitHub
 &lt;<a href="https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/243#issuecomment-169843172" class="issue-link js-issue-link" data-url="https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/243" data-id="125486830" data-error-text="Failed to load issue title" data-permission-text="Issue title is private">#243 (comment)</a>&gt;
 .
</div>
<div class="email-fragment"></div>
</div>
      </div>
    </div>

  </div>

    
<div class="js-socket-channel js-updatable-content comment-reactions  js-reactions-container"
    data-channel="tenant:1:reaction:issue-comment:169843638"
    data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/reactions/issue-comment/169843638">
</div>

</div>

  </div>
  <div class="timeline-comment-wrapper js-comment-container">
    <a href="/AdrianLxM"><img alt="@AdrianLxM" class="timeline-comment-avatar" height="48" src="https://avatars2.githubusercontent.com/u/9692866?v=3&amp;s=96" width="48" /></a>
    <div id="issuecomment-169844890"
     class="comment previewable-edit timeline-comment js-comment js-task-list-container
                    
                     
                    collaborator-comment  "
     data-body-version="98a24bc806a957e2d81c6c9640fc48ab">

  <div class="timeline-comment-header ">

    <div class="timeline-comment-actions">
    </div>

    <span class="timeline-comment-label">Collaborator</span>


  <div class="timeline-comment-header-text">

    <strong>
      <a href="/AdrianLxM" class="author">AdrianLxM</a> 
    </strong>

    commented

    <a href="#issuecomment-169844890" class="timestamp"><relative-time datetime="2016-01-08T00:00:16Z">Jan 8, 2016</relative-time></a>

  </div>
</div>


  <div class="comment-content">

    <div class="edit-comment-hide">
      <div class="comment-body markdown-body markdown-format js-comment-body">
          <p>The same logic as we already have in bgSendQueue as local broadcast - and therefore already show in NightWatch. It is the same as the NightScout-Website shows when enabling raw.</p>
      </div>
    </div>

  </div>

    
<div class="js-socket-channel js-updatable-content comment-reactions  js-reactions-container"
    data-channel="tenant:1:reaction:issue-comment:169844890"
    data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/reactions/issue-comment/169844890">
</div>

</div>

  </div>
  <div class="timeline-comment-wrapper js-comment-container">
    <a href="/tzachi-dar"><img alt="@tzachi-dar" class="timeline-comment-avatar" height="48" src="https://avatars3.githubusercontent.com/u/7942372?v=3&amp;s=96" width="48" /></a>
    <div id="issuecomment-169847784"
     class="comment previewable-edit timeline-comment js-comment js-task-list-container
                    
                     
                    collaborator-comment  "
     data-body-version="3be1360a338283008267138f0b963fa3">

  <div class="timeline-comment-header ">

    <div class="timeline-comment-actions">
    </div>

    <span class="timeline-comment-label">Collaborator</span>


  <div class="timeline-comment-header-text">
      <a href="#issuecomment-169847784" title="This comment left via email reply." class="comment-type-icon">
        <svg aria-hidden="true" class="octicon octicon-mail-read" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path d="M6 5H4V4h2v1zm3 1H4v1h5V6zm5-.48V14c0 .55-.45 1-1 1H1c-.55 0-1-.45-1-1V5.52c0-.33.16-.63.42-.81L2 3.58V3c0-.55.45-1 1-1h1.2L7 0l2.8 2H11c.55 0 1 .45 1 1v.58l1.58 1.13c.27.19.42.48.42.81zM3 7.5L7 10l4-2.5V3H3v4.5zm-2 6l4.5-3-4.5-3v6zm11 .5l-5-3-5 3h10zm1-6.5l-4.5 3 4.5 3v-6z"></path></svg>
      </a>

    <strong>
      <a href="/tzachi-dar" class="author">tzachi-dar</a> 
    </strong>

    commented

    <a href="#issuecomment-169847784" class="timestamp"><relative-time datetime="2016-01-08T00:18:32Z">Jan 8, 2016</relative-time></a>

  </div>
</div>


  <div class="comment-content">

    <div class="edit-comment-hide">
      <div class="comment-body markdown-body email-format js-comment-body">
          <div class="email-fragment">can you point me to some place in the code so I'll be sure about what you
mean?

Thanks
Tzachi</div>
<span class="email-hidden-toggle"><a href="#">…</a></span><div class="email-hidden-reply" style="display:none">
<div class="email-quoted-reply">On Fri, Jan 8, 2016 at 2:00 AM, AdrianLxM ***@***.***&gt; wrote:
 The same logic as we already have in bgSendQueue as local broadcast - and
 therefore already show in NightWatch. It is the same as the
 NightScout-Website shows when enabling raw.

 —
 Reply to this email directly or view it on GitHub
 &lt;<a href="https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/243#issuecomment-169844890" class="issue-link js-issue-link" data-url="https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/243" data-id="125486830" data-error-text="Failed to load issue title" data-permission-text="Issue title is private">#243 (comment)</a>&gt;
 .
</div>
<div class="email-fragment"></div>
</div>
      </div>
    </div>

  </div>

    
<div class="js-socket-channel js-updatable-content comment-reactions  js-reactions-container"
    data-channel="tenant:1:reaction:issue-comment:169847784"
    data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/reactions/issue-comment/169847784">
</div>

</div>

  </div>
  <div class="timeline-comment-wrapper js-comment-container">
    <a href="/AdrianLxM"><img alt="@AdrianLxM" class="timeline-comment-avatar" height="48" src="https://avatars2.githubusercontent.com/u/9692866?v=3&amp;s=96" width="48" /></a>
    <div id="issuecomment-169848450"
     class="comment previewable-edit timeline-comment js-comment js-task-list-container
                    
                     
                    collaborator-comment  "
     data-body-version="e7e70c8c6fe5e455c64cc7fb26decb3d">

  <div class="timeline-comment-header ">

    <div class="timeline-comment-actions">
    </div>

    <span class="timeline-comment-label">Collaborator</span>


  <div class="timeline-comment-header-text">

    <strong>
      <a href="/AdrianLxM" class="author">AdrianLxM</a> 
    </strong>

    commented

    <a href="#issuecomment-169848450" class="timestamp"><relative-time datetime="2016-01-08T00:21:30Z">Jan 8, 2016</relative-time></a>

  </div>
</div>


  <div class="comment-content">

    <div class="edit-comment-hide">
      <div class="comment-body markdown-body markdown-format js-comment-body">
          <p><a href="https://github.com/tzachi-dar" class="user-mention">@tzachi-dar</a>: <a href="https://github.com/nightscout/cgm-remote-monitor/blob/master/lib/plugins/rawbg.js#L59">https://github.com/nightscout/cgm-remote-monitor/blob/master/lib/plugins/rawbg.js#L59</a></p>

<p><a href="https://github.com/mgranberry" class="user-mention">@mgranberry</a>: the android code cleanup also does a lot of things automatically. Unfortunately also some things I don't like. Code duplication just to a very small extent. This one is typical:</p>

<pre><code>diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java b/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java
index dfbc869..d3c8979 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java
@@ -665,10 +665,7 @@ public class EditAlertActivity extends ActivityWithMenu {
             return false;
         }
         Ringtone ringtone = RingtoneManager.getRingtone(context, Uri.parse(path));
-        if(ringtone == null) {
-            return false;
-        }
-        return true;
+        return ringtone != null;
     }

     public String shortPath(String path) {
</code></pre>
      </div>
    </div>

  </div>

    
<div class="js-socket-channel js-updatable-content comment-reactions  js-reactions-container"
    data-channel="tenant:1:reaction:issue-comment:169848450"
    data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/reactions/issue-comment/169848450">
</div>

</div>

  </div>
  <div class="timeline-comment-wrapper js-comment-container">
    <a href="/tzachi-dar"><img alt="@tzachi-dar" class="timeline-comment-avatar" height="48" src="https://avatars3.githubusercontent.com/u/7942372?v=3&amp;s=96" width="48" /></a>
    <div id="issuecomment-169848923"
     class="comment previewable-edit timeline-comment js-comment js-task-list-container
                    
                     
                    collaborator-comment  "
     data-body-version="4cce89745dfffe8ab0eac8865948f9e1">

  <div class="timeline-comment-header ">

    <div class="timeline-comment-actions">
    </div>

    <span class="timeline-comment-label">Collaborator</span>


  <div class="timeline-comment-header-text">
      <a href="#issuecomment-169848923" title="This comment left via email reply." class="comment-type-icon">
        <svg aria-hidden="true" class="octicon octicon-mail-read" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path d="M6 5H4V4h2v1zm3 1H4v1h5V6zm5-.48V14c0 .55-.45 1-1 1H1c-.55 0-1-.45-1-1V5.52c0-.33.16-.63.42-.81L2 3.58V3c0-.55.45-1 1-1h1.2L7 0l2.8 2H11c.55 0 1 .45 1 1v.58l1.58 1.13c.27.19.42.48.42.81zM3 7.5L7 10l4-2.5V3H3v4.5zm-2 6l4.5-3-4.5-3v6zm11 .5l-5-3-5 3h10zm1-6.5l-4.5 3 4.5 3v-6z"></path></svg>
      </a>

    <strong>
      <a href="/tzachi-dar" class="author">tzachi-dar</a> 
    </strong>

    commented

    <a href="#issuecomment-169848923" class="timestamp"><relative-time datetime="2016-01-08T00:24:37Z">Jan 8, 2016</relative-time></a>

  </div>
</div>


  <div class="comment-content">

    <div class="edit-comment-hide">
      <div class="comment-body markdown-body email-format js-comment-body">
          <div class="email-fragment">Is this the answer to my question about raw?</div>
<span class="email-hidden-toggle"><a href="#">…</a></span><div class="email-hidden-reply" style="display:none">
<div class="email-quoted-reply">On Fri, Jan 8, 2016 at 2:21 AM, AdrianLxM ***@***.***&gt; wrote:
 <a href="https://github.com/tzachi-dar" class="user-mention">@tzachi-dar</a> &lt;<a href="https://github.com/tzachi-dar">https://github.com/tzachi-dar</a>&gt;:
 <a href="https://github.com/nightscout/cgm-remote-monitor/blob/master/lib/plugins/rawbg.js#L59">https://github.com/nightscout/cgm-remote-monitor/blob/master/lib/plugins/rawbg.js#L59</a>

 <a href="https://github.com/mgranberry" class="user-mention">@mgranberry</a> &lt;<a href="https://github.com/mgranberry">https://github.com/mgranberry</a>&gt;: the android code cleanup
 also does a lot of things automatically. Unfortunately also some things I
 don't like. This one is typical:

 diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java b/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java
 index dfbc869..d3c8979 100644
 --- a/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java
 +++ b/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java
 @@ -665,10 +665,7 @@ public class EditAlertActivity extends ActivityWithMenu {
              return false;
          }
          Ringtone ringtone = RingtoneManager.getRingtone(context, Uri.parse(path));
 -        if(ringtone == null) {
 -            return false;
 -        }
 -        return true;
 +        return ringtone != null;
      }

      public String shortPath(String path) {

 —
 Reply to this email directly or view it on GitHub
 &lt;<a href="https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/243#issuecomment-169848450" class="issue-link js-issue-link" data-url="https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/243" data-id="125486830" data-error-text="Failed to load issue title" data-permission-text="Issue title is private">#243 (comment)</a>&gt;
 .
</div>
<div class="email-fragment"></div>
</div>
      </div>
    </div>

  </div>

    
<div class="js-socket-channel js-updatable-content comment-reactions  js-reactions-container"
    data-channel="tenant:1:reaction:issue-comment:169848923"
    data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/reactions/issue-comment/169848923">
</div>

</div>

  </div>
  <div class="timeline-comment-wrapper js-comment-container">
    <a href="/AdrianLxM"><img alt="@AdrianLxM" class="timeline-comment-avatar" height="48" src="https://avatars2.githubusercontent.com/u/9692866?v=3&amp;s=96" width="48" /></a>
    <div id="issuecomment-169849415"
     class="comment previewable-edit timeline-comment js-comment js-task-list-container
                    
                     
                    collaborator-comment  "
     data-body-version="f251b9d02e12258707650d658de7846e">

  <div class="timeline-comment-header ">

    <div class="timeline-comment-actions">
    </div>

    <span class="timeline-comment-label">Collaborator</span>


  <div class="timeline-comment-header-text">

    <strong>
      <a href="/AdrianLxM" class="author">AdrianLxM</a> 
    </strong>

    commented

    <a href="#issuecomment-169849415" class="timestamp"><relative-time datetime="2016-01-08T00:26:05Z">Jan 8, 2016</relative-time></a>

  </div>
</div>


  <div class="comment-content">

    <div class="edit-comment-hide">
      <div class="comment-body markdown-body markdown-format js-comment-body">
          <p>Yes, that's the formula. But it's already tested that it's the same we have in BgSendQueue.java.</p>
      </div>
    </div>

  </div>

    
<div class="js-socket-channel js-updatable-content comment-reactions  js-reactions-container"
    data-channel="tenant:1:reaction:issue-comment:169849415"
    data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/reactions/issue-comment/169849415">
</div>

</div>

  </div>
  <div class="timeline-comment-wrapper js-comment-container">
    <a href="/tzachi-dar"><img alt="@tzachi-dar" class="timeline-comment-avatar" height="48" src="https://avatars3.githubusercontent.com/u/7942372?v=3&amp;s=96" width="48" /></a>
    <div id="issuecomment-169849874"
     class="comment previewable-edit timeline-comment js-comment js-task-list-container
                    
                     
                    collaborator-comment  "
     data-body-version="16dcb334f8f6aaa1e2d39c000966c31d">

  <div class="timeline-comment-header ">

    <div class="timeline-comment-actions">
    </div>

    <span class="timeline-comment-label">Collaborator</span>


  <div class="timeline-comment-header-text">
      <a href="#issuecomment-169849874" title="This comment left via email reply." class="comment-type-icon">
        <svg aria-hidden="true" class="octicon octicon-mail-read" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path d="M6 5H4V4h2v1zm3 1H4v1h5V6zm5-.48V14c0 .55-.45 1-1 1H1c-.55 0-1-.45-1-1V5.52c0-.33.16-.63.42-.81L2 3.58V3c0-.55.45-1 1-1h1.2L7 0l2.8 2H11c.55 0 1 .45 1 1v.58l1.58 1.13c.27.19.42.48.42.81zM3 7.5L7 10l4-2.5V3H3v4.5zm-2 6l4.5-3-4.5-3v6zm11 .5l-5-3-5 3h10zm1-6.5l-4.5 3 4.5 3v-6z"></path></svg>
      </a>

    <strong>
      <a href="/tzachi-dar" class="author">tzachi-dar</a> 
    </strong>

    commented

    <a href="#issuecomment-169849874" class="timestamp"><relative-time datetime="2016-01-08T00:28:10Z">Jan 8, 2016</relative-time></a>

  </div>
</div>


  <div class="comment-content">

    <div class="edit-comment-hide">
      <div class="comment-body markdown-body email-format js-comment-body">
          <div class="email-fragment">oops, I was thinking about the ringtone code...</div>
<span class="email-hidden-toggle"><a href="#">…</a></span><div class="email-hidden-reply" style="display:none">
<div class="email-quoted-reply">On Fri, Jan 8, 2016 at 2:26 AM, AdrianLxM ***@***.***&gt; wrote:
 Yes, that's the formula.

 —
 Reply to this email directly or view it on GitHub
 &lt;<a href="https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/243#issuecomment-169849415" class="issue-link js-issue-link" data-url="https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/243" data-id="125486830" data-error-text="Failed to load issue title" data-permission-text="Issue title is private">#243 (comment)</a>&gt;
 .
</div>
<div class="email-fragment"></div>
</div>
      </div>
    </div>

  </div>

    
<div class="js-socket-channel js-updatable-content comment-reactions  js-reactions-container"
    data-channel="tenant:1:reaction:issue-comment:169849874"
    data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/reactions/issue-comment/169849874">
</div>

</div>

  </div>
  <div class="timeline-comment-wrapper js-comment-container">
    <a href="/AdrianLxM"><img alt="@AdrianLxM" class="timeline-comment-avatar" height="48" src="https://avatars2.githubusercontent.com/u/9692866?v=3&amp;s=96" width="48" /></a>
    <div id="issuecomment-169850644"
     class="comment previewable-edit timeline-comment js-comment js-task-list-container
                    
                     
                    collaborator-comment  "
     data-body-version="7d90eb026d3c9e0d86dae0cefde29384">

  <div class="timeline-comment-header ">

    <div class="timeline-comment-actions">
    </div>

    <span class="timeline-comment-label">Collaborator</span>


  <div class="timeline-comment-header-text">

    <strong>
      <a href="/AdrianLxM" class="author">AdrianLxM</a> 
    </strong>

    commented

    <a href="#issuecomment-169850644" class="timestamp"><relative-time datetime="2016-01-08T00:33:20Z">Jan 8, 2016</relative-time></a>

  </div>
</div>


  <div class="comment-content">

    <div class="edit-comment-hide">
      <div class="comment-body markdown-body markdown-format js-comment-body">
          <p>Ah, sorry, I replied do you and to Matthias in one comment. The Ringtone code was an example of what the automatic code cleanup in Android Studio does.</p>
      </div>
    </div>

  </div>

    
<div class="js-socket-channel js-updatable-content comment-reactions  js-reactions-container"
    data-channel="tenant:1:reaction:issue-comment:169850644"
    data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/reactions/issue-comment/169850644">
</div>

</div>

  </div>
  <div class="timeline-comment-wrapper js-comment-container">
    <a href="/tzachi-dar"><img alt="@tzachi-dar" class="timeline-comment-avatar" height="48" src="https://avatars3.githubusercontent.com/u/7942372?v=3&amp;s=96" width="48" /></a>
    <div id="issuecomment-169850703"
     class="comment previewable-edit timeline-comment js-comment js-task-list-container
                    
                     
                    collaborator-comment  "
     data-body-version="7f9ff229010477321ff283fe0c9919bd">

  <div class="timeline-comment-header ">

    <div class="timeline-comment-actions">
    </div>

    <span class="timeline-comment-label">Collaborator</span>


  <div class="timeline-comment-header-text">
      <a href="#issuecomment-169850703" title="This comment left via email reply." class="comment-type-icon">
        <svg aria-hidden="true" class="octicon octicon-mail-read" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path d="M6 5H4V4h2v1zm3 1H4v1h5V6zm5-.48V14c0 .55-.45 1-1 1H1c-.55 0-1-.45-1-1V5.52c0-.33.16-.63.42-.81L2 3.58V3c0-.55.45-1 1-1h1.2L7 0l2.8 2H11c.55 0 1 .45 1 1v.58l1.58 1.13c.27.19.42.48.42.81zM3 7.5L7 10l4-2.5V3H3v4.5zm-2 6l4.5-3-4.5-3v6zm11 .5l-5-3-5 3h10zm1-6.5l-4.5 3 4.5 3v-6z"></path></svg>
      </a>

    <strong>
      <a href="/tzachi-dar" class="author">tzachi-dar</a> 
    </strong>

    commented

    <a href="#issuecomment-169850703" class="timestamp"><relative-time datetime="2016-01-08T00:33:40Z">Jan 8, 2016</relative-time></a>

  </div>
</div>


  <div class="comment-content">

    <div class="edit-comment-hide">
      <div class="comment-body markdown-body email-format js-comment-body">
          <div class="email-fragment">This is very different than what is done in xDrip and I guess that in most
places (expect for dexcom).

I believe that we should stay with the simple formula:

calculated = raw * slope + intercept.

<a href="https://en.wikipedia.org/wiki/Linear_equation">https://en.wikipedia.org/wiki/Linear_equation</a>

[image: Inline image 1]
where *m* is the slope of the line and *b* is the *y* intercept, which is
the *y* coordinate of the location where the line crosses the *y* axis.
This can be seen by letting *x* = 0, which immediately gives *y* = *b*. It
may be helpful to think about this in terms of*y* = *b* + *mx*; where the
line passes through the point (0, *b*) and extends to the left and right at
a slope of *m*. Vertical lines, having undefined slope, cannot be
represented by this form.</div>
<span class="email-hidden-toggle"><a href="#">…</a></span><div class="email-hidden-reply" style="display:none">
<div class="email-quoted-reply">On Fri, Jan 8, 2016 at 2:28 AM, Tzachi Dar ***@***.***&gt; wrote:
 oops, I was thinking about the ringtone code...

 On Fri, Jan 8, 2016 at 2:26 AM, AdrianLxM ***@***.***&gt;
 wrote:

&gt; Yes, that's the formula.
&gt;
&gt; —
&gt; Reply to this email directly or view it on GitHub
&gt; &lt;<a href="https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/243#issuecomment-169849415" class="issue-link js-issue-link" data-url="https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/243" data-id="125486830" data-error-text="Failed to load issue title" data-permission-text="Issue title is private">#243 (comment)</a>&gt;
&gt; .
&gt;

</div>
<div class="email-fragment"></div>
</div>
      </div>
    </div>

  </div>

    
<div class="js-socket-channel js-updatable-content comment-reactions  js-reactions-container"
    data-channel="tenant:1:reaction:issue-comment:169850703"
    data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/reactions/issue-comment/169850703">
</div>

</div>

  </div>
  <div class="timeline-comment-wrapper js-comment-container">
    <a href="/AdrianLxM"><img alt="@AdrianLxM" class="timeline-comment-avatar" height="48" src="https://avatars2.githubusercontent.com/u/9692866?v=3&amp;s=96" width="48" /></a>
    <div id="issuecomment-169853622"
     class="comment previewable-edit timeline-comment js-comment js-task-list-container
                    
                     
                    collaborator-comment  "
     data-body-version="b4efa37e86246ab61c23cee4a7439f2b">

  <div class="timeline-comment-header ">

    <div class="timeline-comment-actions">
    </div>

    <span class="timeline-comment-label">Collaborator</span>


  <div class="timeline-comment-header-text">

    <strong>
      <a href="/AdrianLxM" class="author">AdrianLxM</a> 
    </strong>

    commented

    <a href="#issuecomment-169853622" class="timestamp"><relative-time datetime="2016-01-08T00:47:00Z">Jan 8, 2016</relative-time></a>

  </div>
</div>


  <div class="comment-content">

    <div class="edit-comment-hide">
      <div class="comment-body markdown-body markdown-format js-comment-body">
          <p>This is exactly what we do in xDrip and broadcast to the other apps:<br>
<a href="https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/blob/master/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgSendQueue.java#L128">https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/blob/master/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgSendQueue.java#L128</a></p>

<p>If it makes sense to show the "NightScout-Raw" like this is another discussion, that should be discussed in a separate issue over there: <a href="https://github.com/nightscout/cgm-remote-monitor/">https://github.com/nightscout/cgm-remote-monitor/</a></p>

<p>But the watchface should show the same number that people have on their Nightscout site, their pebbles, the NightWatch watchfaces, ...</p>
      </div>
    </div>

  </div>

    
<div class="js-socket-channel js-updatable-content comment-reactions  js-reactions-container"
    data-channel="tenant:1:reaction:issue-comment:169853622"
    data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/reactions/issue-comment/169853622">
</div>

</div>

  </div>
  <div class="timeline-comment-wrapper js-comment-container">
    <a href="/AdrianLxM"><img alt="@AdrianLxM" class="timeline-comment-avatar" height="48" src="https://avatars2.githubusercontent.com/u/9692866?v=3&amp;s=96" width="48" /></a>
    <div id="issuecomment-169879973"
     class="comment previewable-edit timeline-comment js-comment js-task-list-container
                    
                     
                    collaborator-comment  "
     data-body-version="2dfe2f6f184d74c5b6711dd605cb00a1">

  <div class="timeline-comment-header ">

    <div class="timeline-comment-actions">
    </div>

    <span class="timeline-comment-label">Collaborator</span>


  <div class="timeline-comment-header-text">

    <strong>
      <a href="/AdrianLxM" class="author">AdrianLxM</a> 
    </strong>

    commented

    <a href="#issuecomment-169879973" class="timestamp"><relative-time datetime="2016-01-08T03:05:43Z">Jan 8, 2016</relative-time></a>

  </div>
</div>


  <div class="comment-content">

    <div class="edit-comment-hide">
      <div class="comment-body markdown-body markdown-format js-comment-body">
          <p>To make it more clear that this is slope, intercept, scale and raw as Nightscout understands it and NOT the slope, intercept, ... xDrip internally calculates with:<br>
I extracted methods from the code called <code>getNightscoutSlope()</code>, <code>getNighscoutScale()</code>, ... and put them into <code>NightscoutUploader.java</code> where we already calculated most of that for uploading it to NightScout:<br>
Here is the commit/diff: <a href="https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/commit/542ac5cfb67f5e560f7308dc3fefdbcb4322d1de" class="commit-link"><tt>542ac5c</tt></a></p>
      </div>
    </div>

  </div>

    
<div class="js-socket-channel js-updatable-content comment-reactions  js-reactions-container"
    data-channel="tenant:1:reaction:issue-comment:169879973"
    data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/reactions/issue-comment/169879973">
</div>

</div>

  </div>
  <div class="timeline-comment-wrapper js-comment-container">
    <a href="/tzachi-dar"><img alt="@tzachi-dar" class="timeline-comment-avatar" height="48" src="https://avatars3.githubusercontent.com/u/7942372?v=3&amp;s=96" width="48" /></a>
    <div id="issuecomment-170001014"
     class="comment previewable-edit timeline-comment js-comment js-task-list-container
                    
                     
                    collaborator-comment  "
     data-body-version="c859c8e7a1e01de5293a136940bc61ea">

  <div class="timeline-comment-header ">

    <div class="timeline-comment-actions">
    </div>

    <span class="timeline-comment-label">Collaborator</span>


  <div class="timeline-comment-header-text">
      <a href="#issuecomment-170001014" title="This comment left via email reply." class="comment-type-icon">
        <svg aria-hidden="true" class="octicon octicon-mail-read" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path d="M6 5H4V4h2v1zm3 1H4v1h5V6zm5-.48V14c0 .55-.45 1-1 1H1c-.55 0-1-.45-1-1V5.52c0-.33.16-.63.42-.81L2 3.58V3c0-.55.45-1 1-1h1.2L7 0l2.8 2H11c.55 0 1 .45 1 1v.58l1.58 1.13c.27.19.42.48.42.81zM3 7.5L7 10l4-2.5V3H3v4.5zm-2 6l4.5-3-4.5-3v6zm11 .5l-5-3-5 3h10zm1-6.5l-4.5 3 4.5 3v-6z"></path></svg>
      </a>

    <strong>
      <a href="/tzachi-dar" class="author">tzachi-dar</a> 
    </strong>

    commented

    <a href="#issuecomment-170001014" class="timestamp"><relative-time datetime="2016-01-08T13:19:15Z">Jan 8, 2016</relative-time></a>

  </div>
</div>


  <div class="comment-content">

    <div class="edit-comment-hide">
      <div class="comment-body markdown-body email-format js-comment-body">
          <div class="email-fragment">Here is the way I think it should be, not very far from the way that you
are doing this...

Before I speak of that, let's see how we deal with mgdl and mmol different
units.
Since there is a one to one conversion from one to the other, our live have
been simple.
Xdrip always work in mgdl. Only before printing the data, it might change
that to mmo/l. When data is entered, it is converted from mmo/l to mgdl.
And as I said calculations are always done with mgdl.

On the intercept/slope field, we did not use this method, we have sometimes
been using one of them and sometimes the other, and from here comes the
confusion.
So, I would really like to have all slope intercept used in the xDrip
method and changed only before sending them out.
However, since we have them sometimes stored in one way and sometimes on
the other way, we will probably need to write a convertor in order to do
that. (that is go over our DB and convert from one method to the other when
needed).
Frankly, creating this convertor seems too much for me... so we might end
with the mess we already have.

Does this sound reasonable?

Thanks
Tzachi</div>
<span class="email-hidden-toggle"><a href="#">…</a></span><div class="email-hidden-reply" style="display:none">
<div class="email-quoted-reply">On Fri, Jan 8, 2016 at 5:05 AM, AdrianLxM ***@***.***&gt; wrote:
 To make it more clear that this is slope, intercept, scale and raw as
 Nightscout understands it and NOT the slope, intercept, ... xDrip
 internally calculates with:
 I extracted methods from the code called getNightscoutSlope(),
 getNighscoutScale(), ... and put them into NightscoutUploader.java where
 we already calculated most of that for uploading it to NightScout:
 <a href="https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/commit/542ac5cfb67f5e560f7308dc3fefdbcb4322d1de" class="commit-link"><tt>542ac5c</tt></a>
 &lt;<a href="https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/commit/542ac5cfb67f5e560f7308dc3fefdbcb4322d1de" class="commit-link"><tt>542ac5c</tt></a>&gt;

 —
 Reply to this email directly or view it on GitHub
 &lt;<a href="https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/243#issuecomment-169879973" class="issue-link js-issue-link" data-url="https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/243" data-id="125486830" data-error-text="Failed to load issue title" data-permission-text="Issue title is private">#243 (comment)</a>&gt;
 .
</div>
<div class="email-fragment"></div>
</div>
      </div>
    </div>

  </div>

    
<div class="js-socket-channel js-updatable-content comment-reactions  js-reactions-container"
    data-channel="tenant:1:reaction:issue-comment:170001014"
    data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/reactions/issue-comment/170001014">
</div>

</div>

  </div>
  <div class="timeline-comment-wrapper js-comment-container">
    <a href="/AdrianLxM"><img alt="@AdrianLxM" class="timeline-comment-avatar" height="48" src="https://avatars2.githubusercontent.com/u/9692866?v=3&amp;s=96" width="48" /></a>
    <div id="issuecomment-170233152"
     class="comment previewable-edit timeline-comment js-comment js-task-list-container
                    
                     
                    collaborator-comment  "
     data-body-version="aa984b16fc7129c6ca7f905d0b01efee">

  <div class="timeline-comment-header ">

    <div class="timeline-comment-actions">
    </div>

    <span class="timeline-comment-label">Collaborator</span>


  <div class="timeline-comment-header-text">

    <strong>
      <a href="/AdrianLxM" class="author">AdrianLxM</a> 
    </strong>

    commented

    <a href="#issuecomment-170233152" class="timestamp"><relative-time datetime="2016-01-09T12:30:07Z">Jan 9, 2016</relative-time></a>

  </div>
</div>


  <div class="comment-content">

    <div class="edit-comment-hide">
      <div class="comment-body markdown-body markdown-format js-comment-body">
          <p>Yes, the problem is that we use the same data structure for two types of calibration and distinguish them by the boolean flag <code>check_in</code>. That happened, when the Share support got merged in I guess.<br>
Maybe even earlier as you can add calibrations by connecting a G4 receiver via USB but still use them with transmitter data received by a wixel. This code I didn't touch. It works as it is and changing things here would need extensive care and testing and could break a lot of things.</p>

<p>The methods I created are, as you suggested, just to create data sent to external components (Nightscout and broadcast to other apps that expect it in this format).</p>

<p>Like you said, it sounds reasonable to me as well, not converting the whole base.</p>
      </div>
    </div>

  </div>

    
<div class="js-socket-channel js-updatable-content comment-reactions  js-reactions-container"
    data-channel="tenant:1:reaction:issue-comment:170233152"
    data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/reactions/issue-comment/170233152">
</div>

</div>

  </div>





<!-- Rendered timeline since 2016-01-09 04:30:07 -->
<div id="partial-timeline-marker"
      class="js-timeline-marker js-socket-channel js-updatable-content"
      data-channel="tenant:1:issue:125486830"
      data-url="/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/243/show_partial?partial=issues%2Ftimeline_marker&amp;since=1452342607"
      data-last-modified="Sat, 09 Jan 2016 12:30:07 GMT">

    <!-- </textarea> --><!-- '"` --><form accept-charset="UTF-8" action="/StephenBlackWasAlreadyTaken/xDrip-Experimental/notifications/mark?ids=117628454" class="hidden js-timeline-marker-form" data-form-nonce="2f8075894882f0879ba994800d0f8288ea12e966" data-remote="true" method="post"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="&#x2713;" /><input name="authenticity_token" type="hidden" value="iGOGYA4MjHyeiYSsKsMDJ+hantdkUVK3mgDLT3GRcNBZl2g8EwQQHlQ8x1hbO2gaaUTewZCdFU8IeuIBcTiNEw==" /></div>
</form></div>


        </div>

          <div class="discussion-timeline-actions">
                <div class="signed-out-comment">
    <a href="/join?source=comment-repo" class="btn btn-primary" rel="nofollow">Sign up for free</a>
    <strong>to join this conversation on GitHub</strong>.
    Already have an account?
    <a href="/login?return_to=https%3A%2F%2Fgithub.com%2FStephenBlackWasAlreadyTaken%2FxDrip-Experimental%2Fissues%2F243" rel="nofollow">Sign in to comment</a>
</div>


          </div>
      </div>

    </div>
    <div class="clear"></div>
  </div>

</div>


  </div>
  <div class="modal-backdrop js-touch-events"></div>
</div>


    </div>
  </div>

    </div>

        <div class="container site-footer-container">
  <div class="site-footer" role="contentinfo">
    <ul class="site-footer-links right">
        <li><a href="https://status.github.com/" data-ga-click="Footer, go to status, text:status">Status</a></li>
      <li><a href="https://developer.github.com" data-ga-click="Footer, go to api, text:api">API</a></li>
      <li><a href="https://training.github.com" data-ga-click="Footer, go to training, text:training">Training</a></li>
      <li><a href="https://shop.github.com" data-ga-click="Footer, go to shop, text:shop">Shop</a></li>
        <li><a href="https://github.com/blog" data-ga-click="Footer, go to blog, text:blog">Blog</a></li>
        <li><a href="https://github.com/about" data-ga-click="Footer, go to about, text:about">About</a></li>

    </ul>

    <a href="https://github.com" aria-label="Homepage" class="site-footer-mark" title="GitHub">
      <svg aria-hidden="true" class="octicon octicon-mark-github" height="24" version="1.1" viewBox="0 0 16 16" width="24"><path d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z"></path></svg>
</a>
    <ul class="site-footer-links">
      <li>&copy; 2016 <span title="0.18485s from github-fe152-cp1-prd.iad.github.net">GitHub</span>, Inc.</li>
        <li><a href="https://github.com/site/terms" data-ga-click="Footer, go to terms, text:terms">Terms</a></li>
        <li><a href="https://github.com/site/privacy" data-ga-click="Footer, go to privacy, text:privacy">Privacy</a></li>
        <li><a href="https://github.com/security" data-ga-click="Footer, go to security, text:security">Security</a></li>
        <li><a href="https://github.com/contact" data-ga-click="Footer, go to contact, text:contact">Contact</a></li>
        <li><a href="https://help.github.com" data-ga-click="Footer, go to help, text:help">Help</a></li>
    </ul>
  </div>
</div>



    

    <div id="ajax-error-message" class="ajax-error-message flash flash-error">
      <svg aria-hidden="true" class="octicon octicon-alert" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M8.865 1.52c-.18-.31-.51-.5-.87-.5s-.69.19-.87.5L.275 13.5c-.18.31-.18.69 0 1 .19.31.52.5.87.5h13.7c.36 0 .69-.19.86-.5.17-.31.18-.69.01-1L8.865 1.52zM8.995 13h-2v-2h2v2zm0-3h-2V6h2v4z"></path></svg>
      <button type="button" class="flash-close js-flash-close js-ajax-error-dismiss" aria-label="Dismiss error">
        <svg aria-hidden="true" class="octicon octicon-x" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path d="M7.48 8l3.75 3.75-1.48 1.48L6 9.48l-3.75 3.75-1.48-1.48L4.52 8 .77 4.25l1.48-1.48L6 6.52l3.75-3.75 1.48 1.48z"></path></svg>
      </button>
      Something went wrong with that request. Please try again.
    </div>


      <script crossorigin="anonymous" src="https://assets-cdn.github.com/assets/compat-7db58f8b7b91111107fac755dd8b178fe7db0f209ced51fc339c446ad3f8da2b.js"></script>
      <script crossorigin="anonymous" src="https://assets-cdn.github.com/assets/frameworks-f9a40617e5ea71342dc57103554b07a3c1edc88712b04de1a040aed44891a11d.js"></script>
      <script async="async" crossorigin="anonymous" src="https://assets-cdn.github.com/assets/github-f0c4e3f3e6abd6591d9680dafdae890452bbeda80e6d138134b5c59eaa7cbca3.js"></script>
      
      
      
      
      
      
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-banner hidden">
      <svg aria-hidden="true" class="octicon octicon-alert" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M8.865 1.52c-.18-.31-.51-.5-.87-.5s-.69.19-.87.5L.275 13.5c-.18.31-.18.69 0 1 .19.31.52.5.87.5h13.7c.36 0 .69-.19.86-.5.17-.31.18-.69.01-1L8.865 1.52zM8.995 13h-2v-2h2v2zm0-3h-2V6h2v4z"></path></svg>
      <span class="signed-in-tab-flash">You signed in with another tab or window. <a href="">Reload</a> to refresh your session.</span>
      <span class="signed-out-tab-flash">You signed out in another tab or window. <a href="">Reload</a> to refresh your session.</span>
    </div>
    <div class="facebox" id="facebox" style="display:none;">
  <div class="facebox-popup">
    <div class="facebox-content" role="dialog" aria-labelledby="facebox-header" aria-describedby="facebox-description">
    </div>
    <button type="button" class="facebox-close js-facebox-close" aria-label="Close modal">
      <svg aria-hidden="true" class="octicon octicon-x" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path d="M7.48 8l3.75 3.75-1.48 1.48L6 9.48l-3.75 3.75-1.48-1.48L4.52 8 .77 4.25l1.48-1.48L6 6.52l3.75-3.75 1.48 1.48z"></path></svg>
    </button>
  </div>
</div>

  </body>
</html>

